### PR TITLE
API with file only

### PR DESCRIPTION
**Pull request by Ceredron**
## Description
Removed shipment concept from code

## Related Issue(s)
- #157 

## Verification
- [X] **Your** code builds clean without any errors or warnings
- [X] Manual testing done (required)
- [X] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [X] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Broker/Controllers/FileController.cs [hunk 1]
```diff
@@ -1,105 +1,250 @@
...
 using Altinn.Broker.Mappers;
-using Altinn.Broker.Core.Services.Interfaces;
+using Altinn.Broker.Models;
 using Altinn.Broker.Persistence;
 
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
 namespace Altinn.Broker.Controllers
```
Comment (RagnarFatland-Avanade) [inline src/Altinn.Broker/Controllers/FileController.cs:12]: There seems to be missing a method equivalent to HasAvailableFiles (was HasAvailableShipments), consider implementing it.

File: src/Altinn.Broker.Core/Domain/Enums/ActorFileStatus.cs [hunk 2]
```diff
@@ -1,8 +1,9 @@
 namespace Altinn.Broker.Core.Domain.Enums;
 
-public enum ActorFileStatus {
+public enum ActorFileStatus
+{
     None = 0,
     Initialized = 1,
```
Comment (RagnarFatland-Avanade) [inline src/Altinn.Broker.Core/Domain/Enums/ActorFileStatus.cs:6]: Please update these in separate PR as agreed; https://github.com/Altinn/altinn-broker/issues/164

File: src/Altinn.Broker.Core/Domain/Enums/FileStatus.cs [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
 
-public enum FileStatus {
-    None = 0,
-    Initialized = 1,
-    Processing = 2,
-    Ready = 3,
-    Failed = 4,
-    Deleted = 5
-}
+public enum FileStatus
```
Comment (RagnarFatland-Avanade) [inline src/Altinn.Broker.Core/Domain/Enums/FileStatus.cs:3]: Please update these in separate PR as agreed; https://github.com/Altinn/altinn-broker/issues/164

File: src/Altinn.Broker/Controllers/FileController.cs [hunk 4]
```diff
@@ -1,105 +1,250 @@
...
 
-        [HttpPost]
-        [Route("{fileId}/cancel")]
-        public async Task<ActionResult> CancelFile(Guid fileId, string reasonText)
+        /// <summary>
+        /// Get files that can be accessed by the caller according to specified filters. Result set is limited to 100 files.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        public async Task<ActionResult<List<Guid>>> GetFiles()
```
Comment (RagnarFatland-Avanade) [inline src/Altinn.Broker/Controllers/FileController.cs:170]: Consider adding some simple filters that can be expanded in future;
FileStatus, FromDate, ToDate

File: src/Altinn.Broker.Core/Domain/FileEntity.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+using Altinn.Broker.Core.Domain.Enums;
+
+namespace Altinn.Broker.Core.Domain;
+
+public class FileEntity
+{
+    public Guid FileId { get; set; }
+    public string ApplicationId { get; set; }
+    public string Sender { get; set; } // Joined in
+    public string ExternalFileReference { get; set; }
```
Comment (RagnarFatland-Avanade) [inline src/Altinn.Broker.Core/Domain/FileEntity.cs:10]: Consider renaming this to SendersFileReference to unify the model.

**Review by RagnarFatland-Avanade — CHANGES_REQUESTED** [review]
Looks good, when some of our agreed changes are implemented.

File: src/Altinn.Broker/Models/FileInitializeExt.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        /// Gets or sets the recipients of the broker file.
+        /// </summary>
+        [JsonPropertyName("recipients")]
+        public List<string> Recipients { get; set; } = new List<string>();
+
+        /// <summary>
+        /// Gets or sets the properties field.
+        /// </summary>
+        [JsonPropertyName("metadata")]
+        public Dictionary<string, string> Metadata { get; set; } = new Dictionary<string, string>();
```
Comment (RagnarFatland-Avanade) [inline src/Altinn.Broker/Models/FileInitializeExt.cs:44]: Consider changing the property named "MetaData" to "PropertyList" as this is a less generic name and reflects the naming in the legacy solution for the same purpose. This also avoids confusion with the property "Metadata" in FileInitializeAndUploadExt.

File: src/Altinn.Broker/Controllers/FileController.cs (hunk 4 above)
Comment (Ceredron) [reply src/Altinn.Broker/Controllers/FileController.cs:170]: Added filters, but spun out the actual search algorithm in an independent task:
https://github.com/Altinn/altinn-broker/issues/165

File: src/Altinn.Broker/Controllers/FileController.cs (hunk 1 above)
Comment (Ceredron) [reply src/Altinn.Broker/Controllers/FileController.cs:12]: I will add a corollary to #165 for this

**Pull request merged**